python_modules/dagster/dagster/_core/definitions/asset_condition.py

```diff
@@ -175,65 +217,54 @@ def with_run_ids(self, run_ids: AbstractSet[str]) -> "AssetConditionEvaluationWi
         return AssetConditionEvaluationWithRunIds(evaluation=self, run_ids=frozenset(run_ids))
 
 
-class AssetConditionEvaluationResult(NamedTuple):
-    """Return value for the evaluate method of an AssetCondition."""
+@whitelist_for_serdes
+class AssetConditionEvaluationState(NamedTuple):
+    """Incremental state calculated during the evaluation of an AssetCondition. This may be used
+    on the subsequent evaluation to make the computation more efficient.
+    """
 
-    condition: "AssetCondition"
     evaluation: AssetConditionEvaluation
-    extra_values_by_unique_id: Mapping[str, PackableValue]
+    evaluation_timestamp: Optional[float]
```

Comment:
Is this redundant with the timestamp on `evaluation`?

Reply:
Not exactly. The "evaluation_timestamp" is used as the explicit timestamp for the entire tick's evaluation. This is synchronized across all assets (in theory, this means that you can't get weird situations where we evaluate right at an hour boundary and so two assets with identical partitions definitions were interpreted as having a different set of partitions if one was evaluated before and the other after the hour transition).

This timestamp is used for things like cron schedule ticks as well.

In contrast, the "start_timestamp" on the root evaluation is just when we started evaluating that particular asset (so we can calculate how long that evaluation took).

Reply:
Ah, got it. Would it make sense to call it `previous_tick_evaluation_timestamp` to disambiguate?

Reply:
done!